omegatrade-app/frontend/src/app/components/simulation/simulation.component.ts

```diff
@@ -0,0 +1,148 @@
+import { Component, OnInit, AfterViewInit, ViewChild } from '@angular/core';
+import { MatSort } from '@angular/material/sort';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatTableDataSource } from '@angular/material/table';
+import { SnackBarService } from '../../services/snackbar.service';
+import { FormBuilder, Validators } from '@angular/forms';
+import { RestService } from '../../services/rest.service';
+import { take } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-simulation',
+  templateUrl: './simulation.component.html',
+  styleUrls: ['./simulation.component.css']
+})
+export class SimulationComponent implements OnInit {
+  displayedColumns: string[] = ['companyName', 'companyShortCode', 'status', 'action'];
+  dataSource: MatTableDataSource<SimuationData>;
+  @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+
+  companies: any;
+  simulateForm: any;
+  time = [5, 10, 15, 30, 60];
+  datas = [100, 200, 400, 600];
+  loader: boolean = false;
+  simulations: any;
+  constructor(private snackBarService: SnackBarService, private restService: RestService, private formBuilder: FormBuilder) {
+  }
+
+  ngOnInit(): void {
+    this.getCompanies();
+    this.simulateForm = this.formBuilder.group({
+      companyId: ['', [Validators.required]],
+      timeInterval: ['', [Validators.required]],
+      data: ['', [Validators.required]],
+    });
+    this.getSimulations();
+  }
+
+  getCompanies() {
+    this.restService.getData('companies/list')
+      .pipe(take(1))
+      .subscribe(
+        response => {
+          if (response && response.success) {
+            this.companies = response.data;
+          }
+        },
+        error => {
+          this.snackBarService.openSnackBar(error.error.message, '');
+        });
+  }
+
+  deleteSimulation(row) {
+    this.restService.deleteData(`simulations/delete/${row.sId}`)
```

Comment:
We can take confirmation for deletion from user once, instead of direct deletion,
it can be pressed by mistake,
Also `this.restService.deleteData(`simulations/delete/${simulation.sId}`)` we should check **simulation** existence before accessing child property **simulation.sId**

Reply:
this issue has been resolved

Reply:
Confirmation dialog is done.
But Also this.restService.deleteData(simulations/delete/${simulation.sId}) we should check simulation existence before accessing child property simulation.sId

Reply:
@vignesh-sdev

Reply:
this issue is fixed.